src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,21 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+export const booksSlice = createSlice({
+  name: 'books',
+  initialState: {
+    books: [
+    ],
+  },
+  reducers: {
+    add: (state, action) => {
+      state.books.push(action.payload);
+    },
+    remove: (state, action) => {
+      state.books.filter((book) => book.id !== action.payload);
```

Comment:
- [ ] You did a great job setting up the slices. 👍 Please note that the `filter` array method does not mutate the original array and always returns a new one. Since we want the remove function to update our state we need to assign the result of the filter to our state as follows. Kindly implement it.

```
 state.books = state.books.filter((book) => book.id !== action.payload);
```